File: src/DRandomUniform.java
package UniformImplementation;

import java.security.SecureRandom;
import java.math.BigInteger;

public class DRandomUniform {
  private static SecureRandom r = new SecureRandom();

  public static BigInteger Uniform(BigInteger n) {
    // `longValueExact` will throw an exception if `n` does not fit in a `long`.
    // see https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html#longValueExact--
    return BigInteger.valueOf(r.nextLong(n.longValueExact()));
  }
}

File: src/DRandomUniform.cs
/*******************************************************************************
*  Copyright by the contributors to the Dafny Project
*  SPDX-License-Identifier: MIT
*******************************************************************************/

using System;
using System.Numerics;

namespace UniformImplementation {

    public class DRandomUniform {

      private static Random r = new Random();

      public static BigInteger Uniform(BigInteger n) {
      return new BigInteger(r.Next((int) n));
    }

  }

}
